cloudinit/templater.py

```diff
@@ -42,6 +44,67 @@
 MISSING_JINJA_PREFIX = "CI_MISSING_JINJA_VAR/"
 
 
+class JinjaSyntaxParsingException(TemplateSyntaxError):
+    def __init__(
+        self,
+        error: TemplateSyntaxError,
+        source: str = None,
+    ) -> None:
+        super().__init__(
+            error.message or "unknown syntax error",
+            error.lineno,
+            error.name,
+            error.filename,
+        )
+        self.source = source
+
+    def __str__(self):
+        """Avoid jinja2.TemplateSyntaxErrror multi-line __str__ format."""
+        # line_content = ""
+        # if self.lineno:
+        #     line_content += f" on line {self.lineno}"
+        # if self.source:
+        #     line_content += (
+        #         f": {self.source.splitlines()[self.lineno -1].strip()}"
+        #     )
+        # return (
+        #     f"Unable to parse Jinja template due to syntax error:"
+        #     f" {self.message}{line_content}"
+        # )
+        return self.format_error_message(
+            syntax_error=self.message,
+            line_number=self.lineno,
+            line_content=self.source.splitlines()[self.lineno - 1].strip(),
+        )
+
+    @staticmethod
+    def format_error_message(
+        syntax_error: str,
+        line_number: str,
+        line_content: str = None,
+    ) -> str:
+        """Avoid jinja2.TemplateSyntaxErrror multi-line __str__ format."""
+        template = JinjaSyntaxParsingException.message_template
+        if not line_content:
+            template = template.rsplit(":", 1)[
+                0
+            ]  # remove line_content portion
+            return template.format(
+                syntax_error=syntax_error,
+                line_number=line_number,
+            )
+        return template.format(
```

Comment:
Up here we can set line_content to the appropriate suffix if present or empty string if absent. Then we don't need to manipulate a copy ofthe `message_format` attribute.
```suggestion
        if line_content:
            line_conent = f": {line_content}"
        return template.format(
```